SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -563,10 +563,21 @@ void USpatialNetDriver::BeginDestroy()
 {
 	Super::BeginDestroy();
 
-	// If we are still connected, cleanup our corresponding worker entity if it exists.
+	// If we are still connected...
 	if (Connection != nullptr && WorkerEntityId != SpatialConstants::INVALID_ENTITY_ID)
 	{
-		Connection->SendDeleteEntityRequest(WorkerEntityId);
+		// cleanup our corresponding worker entity if it exists.
+		if (WorkerEntityId != SpatialConstants::INVALID_ENTITY_ID)
+		{
+			Connection->SendDeleteEntityRequest(WorkerEntityId);
+		}
+		
+		// destroy the connection to disconnect from SpatialOS if we aren't meant to persist it.
+		if (!bPersistSpatialConnection)
+		{
+			Connection->DestroyConnection();
```

Comment:
If the net driver is responsible for destroying the connection then the game instance should transfer ownership to it.

Reply:
This would only happen for a client though (if the persist spatial connection flag is false). The server still wants the connection to persist, so it's still the game instance that owns the connection. This shouldn't directly call destroy on it, instead it should tell the game instance to destroy it.

Reply:
This means that ownership is different in the two cases. Unless the net driver asks the game instance to destroy the connection. It does suggest there is a better way of doing this though. As this is another special case to remember and discover when you try to change something.